quickwit-index-config/src/index_object.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use anyhow::{bail, Context};
+use byte_unit::Byte;
+use serde::{Deserialize, Serialize};
+use toml;
+
+use crate::IndexConfig;
+
+#[derive(Serialize, Deserialize)]
+pub struct IndexingResources {
+    pub num_threads: usize,
+    pub heap_size: Byte,
+}
+
+// Is here a good centralized place to define those objects?
```

Comment:
We can import those objects from `quickwit-indexing` but then `quickwit-index-config` depends on `quickwit-indexing`, and `quickwit-indexing` depends on `quickwit-index-config`. With that, `quickwit-metastore` now depends transitively on `quickwit-indexing`. Not great, from a dependency graph perspective. The compiler can figure this out though.